airflow/providers/amazon/aws/transfers/exasol_to_s3.py

```diff
@@ -0,0 +1,117 @@
+from airflow.providers.exasol.hooks.exasol import ExasolHook
+from airflow.utils.decorators import apply_defaults
+
+
+class ExasolToS3Operator(BaseOperator):
+    """
+    Export data from Exasol database to AWS S3 bucket.
+
+    :param query_or_table: the sql statement to be executed or table name to export
+    :type query_or_table: str
+    :param key: S3 key that will point to the file
+    :type key: str
+    :param bucket_name: Name of the bucket in which to store the file
+    :type bucket_name: str
+    :param replace: A flag to decide whether or not to overwrite the key
+        if it already exists. If replace is False and the key exists, an
+        error will be raised.
+    :type replace: bool
+    :param encrypt: If True, the file will be encrypted on the server-side
+        by S3 and will be stored in an encrypted form while at rest in S3.
+    :type encrypt: bool
+    :param gzip: If True, the file will be compressed locally
+    :type gzip: bool
+    :param acl_policy: String specifying the canned ACL policy for the file being
+        uploaded to the S3 bucket.
+    :type acl_policy: str
+    :param query_params: Query parameters passed to underlying ``export_to_file``
+        method of ``ExaConnection``.
+    :type query_params: dict
+    :param export_params: Extra parameters passed to underlying ``export_to_file``
+        method of ``ExaConnection``.
+    :type export_params: dict
+    """
+
+    template_fields = ('query_or_table', 'key', 'bucket_name', 'query_params', 'export_params')
+    template_fields_renderers = {"query_or_table": "sql", "query_params": "json", "export_params": "json"}
+    template_ext = ()
+    ui_color = '#ededed'
+
+    @apply_defaults
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        *,
+        query_or_table: str,
+        key: str,
+        bucket_name: Optional[str] = None,
+        replace: bool = False,
+        encrypt: bool = False,
+        gzip: bool = False,
+        acl_policy: Optional[str] = None,
+        query_params: Optional[Dict] = None,
+        export_params: Optional[Dict] = None,
+        exasol_conn_id: str = 'exasol_default',
+        aws_conn_id: str = 'aws_default',
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.query_or_table = query_or_table
+        self.key = key
+        self.bucket_name = bucket_name
+        self.replace = replace
+        self.encrypt = encrypt
+        self.gzip = gzip
+        self.acl_policy = acl_policy
+        self.query_params = query_params
+        self.export_params = export_params
+        self.exasol_conn_id = exasol_conn_id
+        self.aws_conn_id = aws_conn_id
+
+    def execute(self, context):
+        exasol_hook = ExasolHook(exasol_conn_id=self.exasol_conn_id)
+        s3_hook = S3Hook(aws_conn_id=self.aws_conn_id)
+
+        with NamedTemporaryFile("w+") as file:
+            self.log.info("Downloading data from exasol")
```

Comment:
This line may be unnecessary given you have `self.log.info("Getting data from exasol")` inside `export_to_file `

Reply:
Thanks! Done ✅